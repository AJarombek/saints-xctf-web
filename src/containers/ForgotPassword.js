/**
 * ForgotPassword component for users to change their password when forgotten.
 * @author Andrew Jarombek
 * @since 5/9/2020
 */

import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';

import { userAuthenticated } from '../utils/auth';
import HomeNavBar from '../components/home/HomeNavBar';

const mapStateToProps = state => ({
  auth: state.auth.auth,
  user: state.auth.user
});

const ForgotPassword = ({ auth = {}, user = {} }) => {
  const { signedIn } = auth;
  const history = useHistory();

  useEffect(() => {
    if (userAuthenticated(user, signedIn)) {
      history.push('/dashboard');
    }
  }, [user]);

  return (
    <div className="sxctf-forgot-password">
      <HomeNavBar excludeHeaders={["about", "testimonials"]}/>
    </div>
  );
};

ForgotPassword.propTypes = {
  auth: PropTypes.shape({
    isFetching: PropTypes.bool,
    signedIn: PropTypes.bool,
    status: PropTypes.string
  }),
  user: PropTypes.shape({
    activation_code: PropTypes.string,
    class_year: PropTypes.number,
    deleted: PropTypes.string,
    description: PropTypes.string,
    email: PropTypes.string,
    favorite_event: PropTypes.string,
    first: PropTypes.string,
    last: PropTypes.string,
    last_signin: PropTypes.string,
    location: PropTypes.string,
    member_since: PropTypes.string,
    password: PropTypes.string,
    subscribed: PropTypes.string,
    username: PropTypes.string,
    week_start: PropTypes.string
  })
};

export default connect(mapStateToProps)(ForgotPassword);
