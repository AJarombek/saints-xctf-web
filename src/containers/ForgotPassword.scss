/*
 * Sass styling for the ForgotPassword component
 * @author Andrew Jarombek
 * @since 5/9/2020
 */

.sxctf-forgot-password {

}
