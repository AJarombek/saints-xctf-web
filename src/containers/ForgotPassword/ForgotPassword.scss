/*
 * Sass styling for the ForgotPassword component
 * @author Andrew Jarombek
 * @since 5/9/2020
 */

@import "../../mixins";

.sxctf-forgot-password {
  background-color: #f5f5f5;
}
