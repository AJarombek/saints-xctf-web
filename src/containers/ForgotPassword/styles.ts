/**
 * JSS styles for the ForgotPassword component.
 * @author Andrew Jarombek
 * @since 2/3/2021
 */

export default {
  forgotPassword: {
    backgroundColor: '#f5f5f5'
  }
};
