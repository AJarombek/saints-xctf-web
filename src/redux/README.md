### Overview

Configure Redux as the state management system for the application.

### Files

| Filename                 | Description                                                                |
|--------------------------|----------------------------------------------------------------------------|
| `modules`                | Redux modules containing Reducers, Actions, and Action Creators.           |
| `appState.js`            | Sample application state maintained by Redux.                              |
| `store.js`               | Configuration and creation of the Redux store.                             |
| `types.ts`               | Typescript types used throughout the application.                          |

### Resources

1) [React/Redux Example App #1](https://github.com/erikras/react-redux-universal-hot-example/tree/master/src/redux)
2) [React/Redux Example App #2](https://github.com/goopscoop/ga-react-tutorial/tree/6-reduxActionsAndReducers/src/redux)
3) [React/Redux Example App #3](https://github.com/gothinkster/react-redux-realworld-example-app/tree/master/src)
4) [Redux Ducks Spec](https://github.com/erikras/ducks-modular-redux)
5) [React Router/Redux Setup Steps](https://github.com/supasate/connected-react-router#usage)
6) [Redux Hot Reloading](https://redux.js.org/recipes/configuring-your-store/#hot-reloading)
7) [Redux Thunk](https://github.com/reduxjs/redux-thunk)
8) [Redux/React Official Real World Example](https://github.com/reduxjs/redux/tree/master/examples/real-world)
