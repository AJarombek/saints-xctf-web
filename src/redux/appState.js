/**
 * Sample Redux State for the entire application.
 * @author Andrew Jarombek
 * @since 5/3/2020
 */

const appState = {
  auth: {
    auth: {
      isFetching: false,
      lastUpdated: 1588530309,
      signedIn: true,
      status: "SUCCESS"
    },
    registration: {
      stage: 1,
      first: "Andy",
      last: "Jarombek",
      email: "andrew@jarombek.com",
      username: undefined,
      password: undefined,
      activationCode: undefined,
      teams: {
        saintsXCTFAlumni: {
          status: "accepted",
          groups: {
            alumni: {
              status: "accepted"
            }
          }
        }
      }
    },
    user: {
      andy: {
        isFetching: false,
        didInvalidate: false,
        lastUpdated: 1588530310,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        password: "xxxxxx",
        description: "Test Description",
        memberSince: "2016-02-23 12:00:00",
        classYear: 2017,
        location: "New York, NY",
        favoriteEvent: "Cooldown Jog",
        activationCode: "ABCD1234",
        email: "andrew@jarombek.com",
        lastSignIn: "2020-05-03 12:00:00",
        subscribed: "Y",
        logs: []
      }
    },
  },
  profile: {
    tom: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530311
    },
    joe: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530312
    },
    ben: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530313
    }
  },
  group: {
    wmenstf: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530314,
      groupName: "wmenstf",
      groupTitle: "Women's Track & Field",
      description: "",
      weekStart: "monday",
      members: [],
      logs: []
    }
  },
  logs: {
    isFetching: false,
    didInvalidate: false,
    lastUpdated: 1588530315,
    items: {
      1: {
        log_id: 1,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        name: "Reservoir Loop",
        location: "New York, NY",
        date: "2020-05-03",
        type: "run",
        distance: 6.37,
        metric: "miles",
        miles: 6.37,
        time: "00:45:37",
        pace: "00:07:10",
        feel: 6,
        description: "",
        timeCreated: "2020-05-03 12:00:00",
        comments: {
          1: {
            commentId: 1,
            username: "andy",
            first: "Andrew",
            last: "Jarombek",
            logId: 1,
            time: "2020-05-03 12:00:00",
            content: "test"
          }
        }
      }
    }
  }
};
