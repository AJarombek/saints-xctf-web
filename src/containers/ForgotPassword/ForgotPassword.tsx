/**
 * ForgotPassword component for users to enter their email address to request a
 * "forgot password code".
 *
 * I dont want you to be perfect, I want you to be... you.
 * @author Andrew Jarombek
 * @since 5/9/2020
 */

import React, { useEffect, useRef } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { userAuthenticated } from '../../utils/auth';
import NavBar from '../../components/shared/NavBar';
import ForgotPasswordBody from '../../components/forgot-password/ForgotPasswordBody';
import { RootState } from '../../redux/types';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import { useSetTitle } from '../../hooks/shared';

type Props = {};

const useStyles = createUseStyles(styles);

const ForgotPassword: React.FunctionComponent<Props> = () => {
  useSetTitle('SaintsXCTF');

  const classes = useStyles();

  const navigate = useNavigate();

  const auth = useSelector((state: RootState) => state.auth.auth);
  const users = useSelector((state: RootState) => state.auth.user);

  const ref = useRef(null);

  useEffect(() => {
    if (userAuthenticated(users, auth.signedInUser)) {
      navigate('/dashboard');
    }
  }, [users, auth.signedInUser, navigate]);

  return (
    <div className={classes.forgotPassword} ref={ref}>
      <NavBar includeHeaders={['home', 'register', 'signIn', 'logo']} bodyRef={ref} />
      <ForgotPasswordBody />
    </div>
  );
};

export default ForgotPassword;
