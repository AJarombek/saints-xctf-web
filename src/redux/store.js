/**
 * Build and configure the redux store, which holds the application state and handles
 * all state updates.
 * @author Andrew Jarombek
 * @since 5/1/2020
 */

import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createLogger from 'redux-logger';
import reducer from './modules/reducers';

const history = createBrowserHistory();

const configureStore = (data) => {
  const loggerMiddleware = createLogger();
  const middleware = [loggerMiddleware, routerMiddleware(history)];

  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(reducer, data);
};

export { history };
export default configureStore;
