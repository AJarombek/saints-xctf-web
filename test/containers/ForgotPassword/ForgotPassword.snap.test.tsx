/**
 * Snapshot test for the ForgotPassword component.
 * @author Andrew Jarombek
 * @since 3/11/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { emptyStore } from '../../test-utils/storeMocks';
import { Store } from 'redux';
import ForgotPassword from '../../../src/containers/ForgotPassword';

const mockStore = configureStore([thunk]);

describe('ForgotPassword Snapshot Tests', () => {
  let store: Store;

  beforeEach(() => {
    store = mockStore(emptyStore);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <ForgotPassword />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
