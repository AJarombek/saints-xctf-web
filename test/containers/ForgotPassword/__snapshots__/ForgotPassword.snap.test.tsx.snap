// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Snapshot Tests renders correctly 1`] = `
<div
  className="forgotPassword-0-1-1"
>
  <div
    className="sxctf-nav-bar sxctf-nav-bar-dropdown-hidden dry-0-1-11 lightTheme-0-1-2 lightDropdownHeaderTheme-0-1-7"
  >
    <figure
      className="sxctf-logo"
    >
      <img
        alt=""
        onClick={[Function]}
        src="test-file-stub"
      />
    </figure>
    <h1
      onClick={[Function]}
    >
      SaintsXCTF
    </h1>
    <div
      className="sxctf-nav-buttons"
    >
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-1 ajTextButtonEnabled-2-2-2 homeButton"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Home
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-outlined-button ajOutlinedButton-2-2-7 ajOutlinedButtonEnabled-2-2-8 registerButton"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Register
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-contained-button ajContainedButton-2-2-10 ajContainedButtonEnabled-2-2-11 signInButton"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Sign In
        </button>
      </div>
    </div>
    <div
      className="sxctf-nav-hamburger"
    >
      <div
        className="aj-mobile-hamburger ajMobileHamburger-2-2-13"
        onClick={[Function]}
      >
        <button
          type="button"
        >
          <span
            className="aj-mobile-hamburger-inactive ajMobileHamburgerInactive-2-2-15"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="sxctf-nav-dropdown sxctf-nav-dropdown-hidden dryDropdown-0-1-13 lightDropdownTheme-0-1-5"
  >
    <div
      className="ajNavList-2-2-16 aj-nav-list"
    >
      <div
        className="ajNavListItem-2-2-17 aj-nav-list-item"
      >
        <div
          onClick={[Function]}
        >
          <div>
            Home
          </div>
        </div>
      </div>
      <div
        className="ajNavListItem-2-2-17 aj-nav-list-item"
      >
        <div
          onClick={[Function]}
        >
          <div>
            Register
          </div>
        </div>
      </div>
      <div
        className="ajNavListItem-2-2-17 aj-nav-list-item"
      >
        <div
          onClick={[Function]}
        >
          <div>
            Sign In
          </div>
        </div>
      </div>
      <div
        className="ajNavListItem-2-2-17 aj-nav-list-item ajNavListLastItem-2-2-18 aj-nav-list-last-item"
      >
        <div
          onClick={[Function]}
        >
          <div>
            <img
              alt=""
              className="mobile-dropdown-logo"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sxctf-forgot-password-body"
  >
    <div>
      <h2>
        Forgot Password
      </h2>
      <h5>
        Enter your Email Address or Username to receive a reset code:
      </h5>
      <div
        className="sxctf-image-input"
      >
        <img
          alt=""
          src="test-file-stub"
        />
        <input
          autoComplete=""
          maxLength={50}
          name="email"
          onChange={[Function]}
          placeholder="Email or Username"
          type="text"
        />
        <div>
          <div
            className="status none"
          >
            <p />
          </div>
        </div>
      </div>
      <div
        className="form-buttons"
      >
        <div
          className="ajBaseButton-2-2-4 ajBaseButtonDisabled-2-2-6 aj-contained-button ajContainedButton-2-2-10 ajContainedButtonDisabled-2-2-12 button-0-1-20"
          onClick={null}
        >
          <button
            disabled={true}
            type="button"
          >
            <p>
              Send
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
